### ISSUE TITLE
Plans for documentation

### ISSUE THREAD

**Issue by zordsdavini** [qtile/qtile#1679]
As newcomer or even simple user (as me) documentations is not very intuitive in some parts. I'm not good in documenting but will help where I can. So, notes what looks bad:
- [ ] `Get started`: place to store starting pages about Qtile, install, configuration (this should go into deeper), how to use (this part should be big, with images). _All these sections can be as top level in menu_
- [ ] `Tools` - we should place qshel, iqshel, qtile-top etc. under one menu entry
- [ ] `Command API` and `scripting` (this should have more examples) should go into `Haking on Qtile`
- [ ] `Hacking` content is more about contributing and should be moved here
- [ ] `FAQ` is to small. All answered questions can go here

- [ ] qtile.desktop https://github.com/qtile/qtile/issues/1679#issuecomment-619538578

**Comment by pawamoy** [comment]
+1 on this, except for putting "Command API" and "Scripting" inside of "Hacking on Qtile". Hacking on Qtile is like the common CONTRIBUTING or DEVELOPERS file, it's audience is users who wish to fix bugs, add features, or improve the documentation: forks, development setup and pull requests. Command API and Scripting's audience is users (playing with Qtile features).

**Comment by zordsdavini** [comment]
@pawamoy I disagree with you :) Hacking is for playing with Qtile and what you are saying is hacking should go into Contributing, I think

**Comment by pawamoy** [comment]
I was indeed talking about the contents of "Hacking", not the name :stuck_out_tongue: It seems we agree to say that "Hacking" is not the best name for the current contents (which are more about contributing than hacking/playing).

**Comment by zordsdavini** [comment]
sorry, I didn't read it cearfully. Then I agree with you

**Comment by PeterSR** [comment]
> Get started

I agree. I found this project a couple of days ago. As there was no official package for Ubuntu 20.04, I followed the "Installing From Source", but it wasn't clear how to actually install the window manager such that it could be chosen from the login screen. The answer of course was to add the `qtile.desktop` to `/usr/share/xsessions`, but `pip install qtile` does not do this. 

There is a mention of a `qtile-venv.desktop` at the bottom, but it is stated that it is only if one wants to run Qtile in a virtualenv, so initially I ignored that part. 

It might be pretty easy to piece together the puzzle of finding the `qtile.desktop` in the git repo and adding it, but some newcomers might not find it so easy.

**Comment by PeterSR** [comment]
On my personal wish list is also a short description in `CONTRIBUTING.md` or in the docs about some ideas for how to manage both a stable Qtile and one for adding features, fixing issues etc. The answer is probably just two virtualenvs and two desktop files. 

I am in the process of setting this up myself, so hopefully I can chime in with something when done.

**Comment by zordsdavini** [comment]
@PeterSR http://docs.qtile.org/en/latest/manual/hacking.html#using-xephyr 
I have my own running Qtile and git repo for dev. I just run xephyr to see is all works as expected or `make` for other things like docs, lint, test

**Comment by oboingo** [comment]
@PeterSR I agree that the xsessions section seems like it is related to the venv section, but they are unrelated.  

But I thought the docs were pretty clear on installing...
`pip install .`

I run qtile with the xsession, on Ubuntu 20, but don't have a virtual environment for it (the dependencies are installed in the main OS).  For development though, as @zordsdavini  said, I also use a git repo, with venv, and xephyr.

The documentation could be clearer about this.

**Comment by pawamoy** [comment]
We recently updated the "Hacking on Qtile" documentation page, it should contain everything needed to get ready working on Qtile: http://docs.qtile.org/en/latest/manual/hacking.html. Requirement, venv, Xephyr, debugging, troubleshoot, etc...

**Comment by PeterSR** [comment]
It was not my intention to derail the conversation away from @zordsdavini's initial post. Thank you for the replies!

@oboingo I initially opted for the `pip install qtile` solution to get the latest stable version rather than `pip install .`. Having looked at the `setup.py`, I am pretty sure that neither `pip install` will actually place the `qtile.desktop` in `/usr/share/xsessions` for you. Looking more in the docs, it is however mentioned in Configuration -> Starting Qtile what to do. My point was just to make it more clear in "Installing From Source".

@pawamoy Ah, thanks! There is even a link in `CONTRIBUTING.md`. I must have glossed over it. I think "hacking" to me sounds like something where you create your own neat config, perhaps hacked together, not something were you contribute to an open source project where other people get to see your code.

**Comment by pawamoy** [comment]
@PeterSR 

> I think "hacking" to me sounds like something where you create your own neat config, perhaps hacked together, not something were you contribute to an open source project where other people get to see your code.

Then we are three to think this way! This hacking file should be renamed contributing.

> It was not my intention to derail the conversation away from @zordsdavini's initial post. Thank you for the replies!

I think it's great that everyone is sharing ideas on how to improve the docs, so please continue :smile:

**Comment by superpupervlad** [comment]
[Not important issue]
All descriptions of widgets, layouts and extensions [(for example)](http://docs.qtile.org/en/v0.15.1/manual/ref/widgets.html#libqtile.widget.Image) in brackets " or '. 
While other [descriptions](http://docs.qtile.org/en/latest/manual/commands/index.html#keys) not.

**Comment by m-col** [comment]
Pinning this issue to give it more visibility, as (unsurprisingly) the documentation is often insufficient for people to set up Qtile how they want. Any other ideas/suggestions/criticisms about the docs are very welcome if you've come to this thread because the docs weren't quite enough for you.

**Comment by jeffercbs** [comment]
Me parece genial la iniciativa, no soy una persona muy avanzada y hace poco cuando me lance a configurar mi propio escritorio con Arch Linux y Qtile, Me fue un poco complicado usar qtile. Así que me gustaría ayudar.

**Comment by pawamoy** [comment]
Let me link something that was closed for being stale: #1512

It's about adding auto-generated screenshots of all layouts and their commands. I still believe it would be great to have (animated) pictures in the docs! Moreover the code that generates them was merged, so it should be put to actual use ^^

**Comment by elParaguayo** [comment]
@pawamoy in progress: #3460

**Comment by pawamoy** [comment]
Nice work! Don't hesitate to remove my old code in a subsequent PR. Also, I definitely think that illustrating the layouts commands themselves is very valuable. It will make a *big* doc page, so maybe each layout should be documented on its own page, as I suggested in #1512/#1423. If you go down this road, you can get inspiration for the commands to run from the specs in https://github.com/qtile/qtile/blob/master/docs/screenshots/take_all.py (but I guess you probably read that file already).

**Comment by elParaguayo** [comment]
I have, indeed, seen the code you wrote. It's been very helpful!

**Comment by jeffercbs** [comment]
I think this is a great idea, I recently had some trouble following the documentation for using qtile on my Arch Linux installation. I'm interested in being able to help with this.

**Comment by m-col** [comment]
I had an idea for installation docs: we have a single page with general pip installation and then instead of distro pages, just a list of links to discussions on github. Then we have a discussion for each of the distros that we currently have documented. This way, any issues or changes with their installation will have an obvious *and interactive* place for getting guidance on installing for your distro.

I realise many don't change often, so maybe we could have a hybrid approach where we have a discussion e.g. for Ubuntu installation, while keeping the other pages in the docs.

Is that a good or bad idea? I'm a bit sleep deprived so can't tell if it's actually bad lol.

**Comment by SweepingBishops** [comment]
Why not both? Let's keep the current distro pages as is, and put a link somewhere in them.
"If you're having trouble installing qtile for this particular distro, check out this github discussion."

**Comment by m-col** [comment]
Yep we could do both!

**Comment by yuki-tsubaki** [comment]
I have been working on building a more advanced/comprehensive configuration and while working on it, I ran into a good number of things that didn't have any clear solution any where in the documentation even in the scripting commands layout (e.g. a way to bind a key to activate a particular layout, achievable with the `setlayout` function which is not mentioned anywhere in the documentation). I was also confused on how these scripting functions could be called from in a configuration, and in general how to find things as there was sparse or no information spread across multiple poorly documented points of access.

After this experience, I think that the documentation needs much more comprehensive information on the 'scripting commands' (honestly this feels like a misnomer because they don't have to be used in external scripts).

This should include:
1. More information on how to access them (lazy.function decorators, lazy.object.function decorators lazy.object.function() calls etc.)
2. More information on the commands available through the API (most of them aren't even mentioned). Just having a list of all the available commands would have saved me hours that I spent trying to get information from the shell and later Interactive command client.

I ended up making a script to extract the documentation / command list from the InteractiveCommandClient turned it into a documentation sheet. Something like this is not the best, as even with this method most functions basically only have a function signature, a few have doc strings, and even fewer have examples. However, having the full list of things in a document is much better than having to get things one by one in the shell. I also added a limited example section trying to include access information and examples (point 1). I could upload these if someone wants them (though they aren't finished).

**Comment by elParaguayo** [comment]
Thanks @yuki-tsubaki This is actually something that's very high up the list of our priorities.

You can track progress in #3540 If you have more comments, please add them in there.

**Comment by ervinpopescu** [comment]
@elParaguayo @tych0 @jwijenbergh is this still relevant? I quite like the way docs look now, and https://github.com/qtile/qtile/pull/3540 certainly helped.
